### ISSUE TITLE
Missing Moshi dependency?

### ISSUE THREAD

**Issue by erwan** [DataDog/dd-trace-java#2934]
I recently updated from `dd-trace-ot:0.82.0` to `dd-trace-ot:0.83.2` and now I'm getting a classloader error, apparently Moshi is missing:

```
[error] java.lang.NoClassDefFoundError: com/squareup/moshi/Moshi$Builder
[error] 	at datadog.communication.ddagent.DDAgentFeaturesDiscovery.<clinit>(DDAgentFeaturesDiscovery.java:27)
[error] 	at datadog.trace.common.writer.DDAgentWriter.<init>(DDAgentWriter.java:180)
[error] 	at datadog.trace.common.writer.DDAgentWriter.<init>(DDAgentWriter.java:42)
[error] 	at datadog.trace.common.writer.DDAgentWriter$DDAgentWriterBuilder.build(DDAgentWriter.java:143)
```

And indeed, I noticed that 0.82 used to have a dependency on Moshi: https://mvnrepository.com/artifact/com.datadoghq/dd-trace-ot/0.82.0

...but 0.83 doesn't: https://mvnrepository.com/artifact/com.datadoghq/dd-trace-ot/0.83.2

However Moshi is still used: https://github.com/DataDog/dd-trace-java/blob/master/communication/src/main/java/datadog/communication/ddagent/DDAgentFeaturesDiscovery.java

So where is it supposed to come from? Do we need to include a different library for it to work?

**Comment by mcculls** [comment]
Hi @erwan - yes, a recent refactoring has meant the moshi dependency was missed from the generated Maven pom (this is generated by our gradle build)

You can add this dependency as a temporary workaround
```
<dependency>
<groupId>com.squareup.moshi</groupId>
<artifactId>moshi</artifactId>
<version>1.9.2</version>
</dependency>
```

**Comment by mcculls** [comment]
you may also need to add this dependency
```
  <dependency>
    <groupId>com.datadoghq</groupId>
    <artifactId>java-dogstatsd-client</artifactId>
    <version>2.9.0</version>
  </dependency>
```

meanwhile we'll fix the Maven pom for the next dot release

**Comment by erwan** [comment]
thank you 👍

**Comment by mcculls** [comment]
This has been fixed by #2937 for the next release

**Comment by bantonsson** [comment]
Shouldn't this be kept open until the release is done, to have the automation close it?
